src/vdaf/mastic.rs

```diff
@@ -447,18 +859,125 @@ mod tests {
         thread_rng().fill(&mut verify_key[..]);
         thread_rng().fill(&mut nonce[..]);
 
+        let first_input = VidpfInput::from_bytes(&[240u8, 0u8, 1u8, 4u8][..]);
+        let second_input = VidpfInput::from_bytes(&[112u8, 0u8, 1u8, 4u8][..]);
+        let third_input = VidpfInput::from_bytes(&[48u8, 0u8, 1u8, 4u8][..]);
+        let fourth_input = VidpfInput::from_bytes(&[32u8, 0u8, 1u8, 4u8][..]);
+        let fifth_input = VidpfInput::from_bytes(&[0u8, 0u8, 1u8, 4u8][..]);
+        let first_prefix = VidpfInput::from_bools(&[false, false, true]);
+        let second_prefix = VidpfInput::from_bools(&[false]);
+        let third_prefix = VidpfInput::from_bools(&[true]);
+        let mastic = Mastic::new(algorithm_id, szk, sum_vidpf, 32);
+        let first_agg_param = MasticAggregationParam::new(vec![first_prefix], true).unwrap();
+        let second_agg_param =
+            MasticAggregationParam::new(vec![second_prefix, third_prefix], true).unwrap();
+
+        assert_eq!(
+            run_vdaf(
+                &mastic,
+                &first_agg_param,
+                [
+                    (first_input.clone(), true),
+                    (second_input.clone(), false),
+                    (third_input.clone(), false),
+                    (fourth_input.clone(), true),
+                    (fifth_input.clone(), true)
+                ]
+            )
+            .unwrap(),
+            vec![1]
+        );
+
+        assert_eq!(
+            run_vdaf(
+                &mastic,
+                &second_agg_param,
+                [
+                    (first_input.clone(), true),
+                    (second_input, false),
+                    (third_input, false),
+                    (fourth_input, true),
+                    (fifth_input, true)
+                ]
+            )
+            .unwrap(),
+            vec![2, 1]
+        );
+
+        let (public_share, input_shares) =
+            mastic.shard(&(first_input.clone(), false), &nonce).unwrap();
+        run_vdaf_prepare(
+            &mastic,
+            &verify_key,
+            &first_agg_param,
+            &nonce,
+            public_share,
+            input_shares,
+        )
+        .unwrap();
+
+        let (public_share, input_shares) = mastic.shard(&(first_input, true), &nonce).unwrap();
+        run_vdaf_prepare(
+            &mastic,
+            &verify_key,
+            &first_agg_param,
+            &nonce,
+            public_share,
+            input_shares,
+        )
+        .unwrap();
```

Comment:
I think we should use `run_vdaf()` to run the whole thing end-to-end, including aggregation and collection: https://github.com/divviup/libprio-rs/blob/d2fe428cadaf690a56a74f4c682f62326f959f06/src/vdaf.rs#L486